File: erpwebapp.client/src/app/booking/booking/interfaces/booking.ts
import { Car } from '../../car/interfaces/car';

export interface BookingUser {
  id: string;
  firstName: string;
  lastName: string;
  email?: string | null;
}

export interface Booking {
  id: number;
  bookingNumber: string;
  userId: string;
  userName: string;
  carId: number;
  carName: string;
  pickupDate: string;
  returnDate: string;
  totalDays: number;
  amount: number;
  status: string;
  paymentStatus: string;
  createdDate: string;
}

export type BookingDialogMode = 'create' | 'edit' | 'view';

export interface BookingDialogData {
  mode: BookingDialogMode;
  booking?: Booking;
  cars: Car[];
  users: BookingUser[];
}

File: ERPWebAppData/Entity/Booking.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using WebApp.Data.Entity;

namespace ERPWebAppData.Entity
{
    public class Booking
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(30)]
        public string BookingNumber { get; set; } = string.Empty;

        [Required]
        public string UserId { get; set; } = string.Empty;

        [Required]
        public int CarId { get; set; }

        public DateTime PickupDate { get; set; }

        public DateTime ReturnDate { get; set; }

        public int TotalDays { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        [MaxLength(20)]
        public string Status { get; set; } = "Pending";

        [MaxLength(20)]
        public string PaymentStatus { get; set; } = "Pending";

        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

        // Navigation Properties
        public virtual Car? Car { get; set; }

        [System.Text.Json.Serialization.JsonIgnore]
        public virtual User? User { get; set; }
    }
}
